inc/REST/AuthController.php

```diff
@@ -68,11 +83,44 @@ public function get_google_auth_token( WP_REST_Request $request ): WP_REST_Respo
 		);
 	}
 
+	public function get_salesforce_d2c_stores( WP_REST_Request $request ): WP_REST_Response|WP_Error {
+		$params = $request->get_json_params();
+		$client_id = $params['clientId'] ?? null;
+		$client_secret = $params['clientSecret'] ?? null;
+		$domain = $params['domain'] ?? null;
+
+		if ( ! $client_id || ! $client_secret || ! $domain ) {
+			return new \WP_Error(
+				'missing_parameters',
+				__( 'Client ID, client secret and domain are required.', 'remote-data-blocks' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		$endpoint = 'https://' . $domain . '.my.salesforce.com';
+
+		$token = SalesforceD2CAuth::generate_token( $endpoint, $client_id, $client_secret );
+		if ( is_wp_error( $token ) ) {
+			return rest_ensure_response( $token );
```

Comment:
Are we sure these errors don't leak anything sensitive?

Reply:
This is something I meant to remove before the final PR as it was only in place for testing. I'll take it out and replace it with a generic message.